core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -156,7 +157,32 @@ private[spark] class PipedRDD[T: ClassTag](
           out.close()
         }
       }
-    }.start()
+    }
+    stdinWriterThread.start()
+
+    // interrupts stdin writer and stderr read threads when the corresponding task is finished as a
+    // safe belt. Otherwise, these threads could outlive the task's lifetime. For example:
+    //   val pipeRDD = sc.range(1, 100).pipe(Seq("cat"))
+    //   val abnormalRDD = pipeRDD.mapPartitions(_ => Iterator.empty)
+    // the iterator generated by PipedRDD is never involved. If the parent RDD's iterator is time
+    // consuming to generate(ShuffledRDD's shuffle operation for example), the outlived stdin writer
+    // thread will consume significant memory and cpu time. Also, there's race condition for
```

Comment:
Does the second part of the comment, beginning at "Also,", belong below in the change to `ShuffleBlockFetcherIterator`?

Reply:
This is a tricky one. After the fix in this pr to `ShuffleBlockFetcherIterator`, the race condition should be really rare as only one potential next call may hang.  However I am unable to find a good place to put the above comment in `ShuffleBlockFetcherIterator`. So it's inside this `task completion listener`.

Do you have any suggestion?

Reply:
Why not put the comment next to the change in ShuffleBlockFetcherIterator?